src/ndv/models/_roi_model.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pydantic import field_validator
+
+from ndv.models._base_model import NDVModel
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+class RectangularROIModel(NDVModel):
+    """Representation of an axis-aligned rectangular Region of Interest (ROI).
+
+    Attributes
+    ----------
+    visible : bool
+        Whether to display this roi.
+    bounding_box : tuple[tuple[float, float], tuple[float, float]]
+        The minimum (2D) point and the maximum (2D) point contained within the
+        region. Using these two points, an axis-aligned bounding box can be
+        constructed.
+    """
+
+    visible: bool = True
+    bounding_box: tuple[tuple[float, float], tuple[float, float]] = ((0, 0), (0, 0))
+
+    @field_validator("bounding_box", mode="after")
+    @classmethod
+    def _validate_bounding_box(
+        cls, bb: Any
+    ) -> tuple[tuple[float, float], tuple[float, float]]:
+        if not isinstance(bb, tuple):
+            raise ValueError(f"{bb} not a tuple of points!")
```

Comment:
```suggestion
        cls, bb: tuple[tuple[float, float], tuple[float, float]]
    ) -> tuple[tuple[float, float], tuple[float, float]]:
```

(because mode='after')